### PR TITLE
Standardize search/find to search

### PR DESCRIPTION
**Pull request by hius07**
There is `File search` in the Gesture manager already.
There is `Fulltext search` in the readermenu.

Some help text added.

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/7398)

### CODE REVIEW

File: frontend/apps/filemanager/filemanagermenu.lua [hunk 1]
```diff
@@ -612,7 +612,13 @@ To:
 
     self.menu_items.find_file = {
         -- @translators Search for files by name.
-        text = _("Find a file"),
+        text = _("File search"),
+        help_text = _([[Search a book by filename in the current or home folder and its subfolders.
+
+Search for '*' will show all files.
```
Comment (NiLuJe) [inline frontend/apps/filemanager/filemanagermenu.lua:618]:
```suggestion
A search for '*' will show all files.
```

**Comment by Frenzie** [comment]
NB

Screenshot_2021-03-08-09-05-47-21_1aff61e6bd981aaa4ba9d0245eb6c04f

**Comment by hius07** [comment]
> NB

`Calibre search` would be clear and unified.
https://github.com/koreader/koreader/blob/0674e6060aca0b88ea76b3b056db3918ccd71786/plugins/calibre.koplugin/main.lua#L112

**Comment by Frenzie** [comment]
I would keep `Find a file` and change it to `Find in text`.

**Comment by hius07** [comment]
Not a simple question)
https://www.dictionary.com/browse/find
https://www.dictionary.com/browse/search
https://www.differencebetween.com/difference-between-search-and-vs-find-and-vs-seek/
Chrome ctrl+f description: 'Open the Find Bar to search the current page"

Based on the definition, `Find` seems to be a little bit more accurate.
In the meantime, there are many `Search results` window headers. Can it be unified somehow?

**Comment by Frenzie** [comment]
The equivalent with search would be something like this:

* Search for a file
* Search in text

> In the meantime, there are many Search results window headers. Can it be unified somehow?

Most things in the menu are very much on purpose. Plenty of headers are legacy.

**Comment by hius07** [comment]
Kindle uses `Search` everywhere: in text, book names on device, Kindle Store.

These are the similar activities (from a user's point of view):
`Text search`
`File search`
`Calibre search`
`Wikipedia search`
`Dictionary search`
And all of them give `Search results`.



Find, search, lookup, look up - the Occam's razor is needed.

**Comment by poire-z** [comment]
(No real opinion about these - I think I'd like this uniformity, but I don't know if we should really go toward radical uniformity - and impoverish vocabulary and have users learn/get used to poor english expressions :) From French, I would naturally use "search a word in a dict" - but it seems to me that "look up a word in a dict" feels like better english - and I might now use that after seeing it that often in KOReader :)

**Comment by Frenzie** [comment]
```
Text search
File search
Calibre search
Wikipedia search
Dictionary search
```

Find/search is fairly immaterial. In my mind `Find` is the canonical KOReader one. But generally it should be an action.

Find/search in text
Find/search for (a) file
Find/search in calibre catalog
Find/search in Wikipedia
Find/search in dictionary

Incidentally, the Apple dictionary app uses all three. 

* Type a word to look up in…
* Search
* Find → Search For a New Word…

**Comment by hius07** [comment]
We can see the following entries:

1 **Text**
1.1 `Fulltext search` main menu
1.2 `Enter text to search for` dialog header
1.3 `Search` button in the context (highlight) menu
1.4 `Fulltext search` dispatcher

2 **File**
2.1 `Find a file` main menu
2.2 `Search for books by filename` dialog header
2.3 `Search Results` window header
2.4 `File search` dispatcher

3 **OPDS catalog**
3.1 `Search` everywhere

4 **Dictionary**
4.1 `Dictionary lookup` main menu
4.2 `Enable fuzzy search` dictionary settings
4.3 `Enter a word to look up` main dailog header
4.4 `Search dictionary` main dialog button
4.5 `Search` dictionary article button
4.6 `Input lookup word` additional dialog header
4.7 `Lookup` additional dialog button
4.8 `Dictionary lookup` dispatcher

5 **Wikipedia**
5.1 `Wikipedia lookup` main menu
5.2 `Show image in search results` Wikipedia settings
5.3 `Enter words to look up on Wikipedia` main dailog header
5.4 `Search Wikipedia` main dialog button
5.5 `Searching Wikipedia` infomessage
5.6 `Input lookup word` additional dialog header
5.7 `Lookup` additional dialog button
5.8 `Wikipedia lookup` dispatcher

6 **Calibre**
6.1 `Find a book via calibre metadata` main menu
6.2 `Search settings` tools Calibre menu, 5 more `search` entries within
6.3 `Search in calibre metadata` dispatcher

To summarize:
1. `Find` is used in the main menu for Files and Calibre search only (2 entries).
2. `Lookup`/`look up` is used in the main menu and one dialog for Dictionary and Wikipedia.
3. `Search` is used for all settings and results in Dictionary and Wikipedia search.
4. `Search` is prevailing mostly everywhere (so far).

Hope this helps you, the code owners, to make the final decision on terminology unification.

**Comment by hius07** [comment]
I would be happy to provide further unification based on two decisions:
- "Search" or "Find"?
- keep "lookup" for Wiki and Dictionary?

**Comment by poire-z** [comment]
I don't use the 2 "Find..." so I won't notice the change - I'm fine with the current "Search" ones.
I'm also fine with the "lookup" where it's used (the switch to "search" feels not necessary to me - may be just because I'm used to them :)

**Comment by hius07** [comment]
`Search` - a noun and a verb, `lookup` - a noun, `Look up` - a verb.
`Enter smth to search for` in all InputDialog titles.
`No results` everywhere when nothing found.

Sorry, don't understand a conflict with opdsbrowser.lua.

File: frontend/apps/reader/modules/readerdictionary.lua [hunk 2]
```diff
@@ -678,7 +677,7 @@ end
 
 function ReaderDictionary:onShowDictionaryLookup()
     self.dictionary_lookup_dialog = InputDialog:new{
-        title = _("Enter a word to look up"),
+        title = _("Enter word(s) to look up"),
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readerdictionary.lua:680]: I don't like this change. It's very unnatural.

File: plugins/calibre.koplugin/main.lua [hunk 3]
```diff
@@ -59,9 +59,9 @@ function Calibre:closeWirelessConnection()
 end
 
 function Calibre:onDispatcherRegisterActions()
-    Dispatcher:registerAction("calibre_search", { category="none", event="CalibreSearch", title=_("Search in calibre metadata"), device=true,})
-    Dispatcher:registerAction("calibre_browse_tags", { category="none", event="CalibreBrowseTags", title=_("Browse all calibre tags"), device=true,})
-    Dispatcher:registerAction("calibre_browse_series", { category="none", event="CalibreBrowseSeries", title=_("Browse all calibre series"), device=true, separator=true,})
+    Dispatcher:registerAction("calibre_search", { category="none", event="CalibreSearch", title=_("Calibre search"), device=true,})
```
Comment (Frenzie) [inline plugins/calibre.koplugin/main.lua:62]: Some calibre nitpickers will be upset.

Comment (Frenzie) [reply plugins/calibre.koplugin/main.lua:62]: Concretely, I mean it's purposefully calibre. That's not a typo or a mistake.

**Review by Frenzie — COMMENTED** [review]
Thanks!

File: plugins/opds.koplugin/opdsbrowser.lua [hunk 4]
```diff
@@ -739,7 +737,7 @@ function OPDSBrowser:browseSearchable(browse_url, username, password)
 end
 
 function OPDSBrowser:onMenuSelect(item)
-    self.catalog_title = item.text or _("OPDS Catalog")
+    self.catalog_title = item.text or _("OPDS catalog")
```
Comment (Frenzie) [inline plugins/opds.koplugin/opdsbrowser.lua:740]: I resolved it but I can't seem to push to your repo. Did you disable push access for repo owners when creating the PR?

Anyway, the easiest way to resolve in lieu of command line git is to undo this change:

```suggestion
    self.catalog_title = item.text or _("OPDS Catalog")
```

Then click something like "merge with master" which will probably be an option in the PR once you do that to bring it up to date, and re-apply on the current code.

Comment (Frenzie) [reply plugins/opds.koplugin/opdsbrowser.lua:740]: See https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

File: frontend/apps/reader/modules/readerdictionary.lua (hunk 2 above)
Comment (hius07) [reply frontend/apps/reader/modules/readerdictionary.lua:680]: I can look up `upside down` (2 words) in English-Russian dictionary and get results.

File: plugins/calibre.koplugin/main.lua (hunk 3 above)
Comment (hius07) [reply plugins/calibre.koplugin/main.lua:62]: Sorry, you are absolutely right, this should be reverted.

File: frontend/apps/reader/modules/readerdictionary.lua (hunk 2 above)
Comment (NiLuJe) [reply frontend/apps/reader/modules/readerdictionary.lua:680]: True, but the *wording* is weird ;).

The same meaning might be able to come across another way, possibly something based on "search term", which is understood to mean word/words/phrase ;).

Comment (hius07) [reply frontend/apps/reader/modules/readerdictionary.lua:680]: "Query" is perfect, it is duplicated in the lookup results window.
`Enter search query` for title, ok-button remains `Look up`.

Comment (Frenzie) [reply frontend/apps/reader/modules/readerdictionary.lua]: > I can look up upside down (2 words) in English-Russian dictionary and get results.

The proper term is `lemma` but the better solution is `word or phrase`.

File: plugins/calibre.koplugin/main.lua [hunk 5]
```diff
@@ -109,7 +109,7 @@ function Calibre:addToMainMenu(menu_items)
     -- insert the metadata search
     if G_reader_settings:isTrue("calibre_search_from_reader") or not self.ui.view then
         menu_items.find_book_in_calibre_catalog = {
-            text = _("Find a book via calibre metadata"),
+            text = _("Calibre search"),
```
Comment (Frenzie) [inline plugins/calibre.koplugin/main.lua:112]: While I lean more toward "our style guide says to start a phrase or fragment with an uppercase character," I think `calibre search` would also be included in that.

@NiLuJe probably has a better finger on the calibre pulse.

File: frontend/ui/widget/dictquicklookup.lua [hunk 6]
```diff
@@ -1186,7 +1186,7 @@ end
 
 function DictQuickLookup:lookupInputWord(hint)
     self.input_dialog = InputDialog:new{
-        title = _("Input lookup word"),
+        title = _("Enter search query"),
```
Comment (Frenzie) [inline frontend/ui/widget/dictquicklookup.lua:1189]: "Enter search query" is very Computer English. I'm not a fan.
```suggestion
        title = _("Enter a word or phrase to look up"),
```

File: plugins/calibre.koplugin/main.lua (hunk 5 above)
Comment (NiLuJe) [reply plugins/calibre.koplugin/main.lua:112]: There's a bit of a catch-22 in that the official spelling of the app is actually `calibre` (lowercase), which has a tendency to drive me mildly nuts ;p.

Comment (NiLuJe) [reply plugins/calibre.koplugin/main.lua:112]: In any case, in this context, I feel like that would be a bit *too* short, anyway ;).

And people are already confused enough that I think the current spelling is perfectly fine?

File: frontend/apps/reader/modules/readerdictionary.lua [hunk 7]
```diff
@@ -690,7 +689,7 @@ function ReaderDictionary:onShowDictionaryLookup()
                     end,
                 },
                 {
-                    text = _("Search dictionary"),
+                    text = _("Look up"),
```
Comment (NiLuJe) [inline frontend/apps/reader/modules/readerdictionary.lua:692]: `Lookup` ^^.

I'm half of a mind to actually keep it simple and just use `Search`, actually.

File: frontend/apps/reader/modules/readerwikipedia.lua [hunk 8]
```diff
@@ -49,7 +49,7 @@ function ReaderWikipedia:lookupInput()
                     end,
                 },
                 {
-                    text = _("Search Wikipedia"),
+                    text = _("Look up"),
```
Comment (NiLuJe) [inline frontend/apps/reader/modules/readerwikipedia.lua:52]: Ditto

File: frontend/ui/widget/dictquicklookup.lua [hunk 9]
```diff
@@ -1199,7 +1199,7 @@ function DictQuickLookup:lookupInputWord(hint)
                     end,
                 },
                 {
-                    text = _("Lookup"),
+                    text = _("Look up"),
```
Comment (NiLuJe) [inline frontend/ui/widget/dictquicklookup.lua:1202]: Ditto

File: plugins/opds.koplugin/opdsbrowser.lua [hunk 10]
```diff
@@ -706,13 +706,11 @@ end
 
 function OPDSBrowser:browseSearchable(browse_url, username, password)
     self.search_server_dialog = InputDialog:new{
-        title = _("Search OPDS catalog"),
+        title = _("Enter author or title to search for"),
```
Comment (NiLuJe) [inline plugins/opds.koplugin/opdsbrowser.lua:709]: A bit reductive, you can arguably match on tags in a Calibre catalog.

(i.e. , I prefer the previous title + hint combo, possibly with minor tweaks).

**Comment by hius07** [comment]
What do you think about unified lookup input dialog?


It's just a prototype, not a working code.

File: plugins/opds.koplugin/opdsbrowser.lua (hunk 10 above)
Comment (hius07) [reply plugins/opds.koplugin/opdsbrowser.lua:709]: > I prefer the previous title + hint combo

Do you mean `input_hint`?
There is just a `hint` in the code
https://github.com/koreader/koreader/blob/6d6760c2b3530b1469f127e6c06a3b49b3e1eca5/plugins/opds.koplugin/opdsbrowser.lua#L708
but I cannot find it in the InputDialog so I've removed it.

File: plugins/calibre.koplugin/main.lua (hunk 5 above)
Comment (Frenzie) [reply plugins/calibre.koplugin/main.lua:112]: > In any case, in this context, I feel like that would be a bit too short, anyway ;).

You mean it should be at least `calibre metadata search` or just no change at all?

File: frontend/apps/reader/modules/readerdictionary.lua (hunk 7 above)
Comment (Frenzie) [reply frontend/apps/reader/modules/readerdictionary.lua:692]: No, `look up` the phrasal verb (as in `click look up to perform a lookup`).

> I'm half of a mind to actually keep it simple and just use Search, actually.

I'd prefer to just keep it `Search dictionary`.

File: frontend/apps/reader/modules/readerwikipedia.lua (hunk 8 above)
Comment (Frenzie) [reply frontend/apps/reader/modules/readerwikipedia.lua:52]: Ditto indeed, I prefer Search Wikipedia.

**Comment by Frenzie** [comment]
> What do you think about unified lookup input dialog?

I generally prefer simpler dialogs where possible, with a simple and obvious left = cancel, right = the action, without any middle.

That way the menu may be more complex but the dialog is simpler. It's a trade-off.

**Comment by hius07** [comment]
Thank you.
The reason is that this input dialog is the same in all 3 modules: Dict, Wiki and search results DictQuickLookup.
BTW, what ok_button is recommended in DictQuickLookup (where ok_button action depends on the previous search)?

Now we have a very-long-press easter egg for switching between Dict and Wiki
https://github.com/koreader/koreader/blob/6d6760c2b3530b1469f127e6c06a3b49b3e1eca5/frontend/ui/widget/dictquicklookup.lua#L161
so the idea of domain switching is in the air.
Three buttons allow simple switching and show unified accustomed dialogs.

If not agreed, as a compromise it might be
`Search Dictionary`, `Search Wikipedia` and three buttons in DictQuickLookup.

**Comment by Frenzie** [comment]
> The reason is that this input dialog is the same in all 3 modules: Dict, Wiki and search results DictQuickLookup.

I suspect it's not (except visually), but that's an interesting point.

**Comment by hius07** [comment]
> except visually

Yeah, it's hard for a developer to stand at the average user's point of view. He knows what's under the hood.

**Comment by hius07** [comment]
> what ok_button is recommended in DictQuickLookup (where ok_button action depends on the previous search)?

Answer myself: we can have the same `Dict/Wiki search` button, depending on domain.

**Comment by hius07** [comment]
Since this PR is about wordings only, let's postpone redesign of the search input.
If anybody likes three buttons (for the purpose of simple domain switching), please let me know, I'll make the propositions.

File: plugins/opds.koplugin/opdsbrowser.lua [hunk 11]
```diff
@@ -706,13 +706,11 @@ end
...
         input = "",
-        hint = _("Search string"),
         -- @translators: This is an input hint for something to search for in an OPDS catalog, namely a famous author everyone knows. It probably doesn't need to be localized, but this is just here in case another name or book title would be more appropriate outside of a European context.
         input_hint = _("Alexandre Dumas"),
         input_type = "string",
-        description = _("%s in url will be replaced by your input"),
```
Comment (Frenzie) [inline plugins/opds.koplugin/opdsbrowser.lua:715]: This strikes me as rather important information?
Screenshot_2021-04-02_13-15-50

Comment (hius07) [reply plugins/opds.koplugin/opdsbrowser.lua:715]: Assume I am not an IT man.
What is `%s`? In what url? I just want to search for Dan Brown.
And even if I understand the above, does it change my input?

Comment (Frenzie) [reply plugins/opds.koplugin/opdsbrowser.lua:715]: Hold on, let me check the git blame for this one. I'm not very much at home in the OPDS plugin. Indeed it seems somewhat random here. But the point remains that it's important information *somewhere*. If it's here rather than where it should be, that may be an accident that should be investigated rather than removed wholesale.

If you add a URL like `https://www.bing.com/search?FORM=INCOH2&PC=IFJ1&PTAG=ICO-c9d0fc87&q=%s` then `%s` will be the encoded search terms. Leave out the %s and you have no search.

Comment (Frenzie) [reply plugins/opds.koplugin/opdsbrowser.lua:715]: Context: https://github.com/koreader/koreader/pull/5823

**Comment by poire-z** [comment]
> If anybody likes three buttons (for the purpose of simple domain switching), please let me know, I'll make the propositions.

When would we see it? Only when hitting the little pencil icon at top right of any lookup result (that I can figure from you screenshot), or elsewhere too?

File: plugins/opds.koplugin/opdsbrowser.lua (hunk 11 above)
Comment (hius07) [reply plugins/opds.koplugin/opdsbrowser.lua:715]: I remember it was redesigned. But...
https://github.com/koreader/koreader/pull/7380#issuecomment-792099693

**Comment by hius07** [comment]
> When would we see it?

As we have one combined search results window (DictQuickLookup), we can have one combined input invoked from the main menu. Calling it from the DictQuickLookup allows switching domans for a new search.

**Comment by hius07** [comment]
I'm done with wordings, all reprimands fixed. Thank you very much.

**Comment by poire-z** [comment]
So, the little pencil icon + the menu. Is that all ?

When you say "main menu", I assume you mean the 2 entries `Dictionary lookup` and `Wikipedia lookup` , right ?

If these 2 would become a single menu item `Lookup`, it would be a little less clear in this menu, as we currently have 3 items related to dictionary, a separator, 3 same items related to Wikipedia.

Or do you see us keeping that existing menu, and both "lookup" items launching that same dialog with both buttons ? (If yes, do you see these buttons inverted, so the one related to the menu item hit is the right most one?)

**Comment by hius07** [comment]
> So, the little pencil icon + the menu. Is that all ?

Yes, I thought about these 3 modules and to have just one main menu entry for lookup.
Let me think more over it, I see that the advantages (besides of input dialog unification) are no so obvious.

File: plugins/calibre.koplugin/main.lua (hunk 5 above)
Comment (NiLuJe) [reply plugins/calibre.koplugin/main.lua:112]: `calibre metadata search` would work, FWIW, but I also have no issue with the current wording ;).

File: plugins/opds.koplugin/opdsbrowser.lua (hunk 10 above)
Comment (NiLuJe) [reply plugins/opds.koplugin/opdsbrowser.lua]: Possibly there was a typo in the field name, yeah, but I know the feature exists ;p.

File: plugins/opds.koplugin/opdsbrowser.lua (hunk 11 above)
Comment (NiLuJe) [reply plugins/opds.koplugin/opdsbrowser.lua]: IIRC, it puzzled me, and I checked, and I *think* it still makes sense is some edge-cases w/ legacy searches.

But it's weird.

Comment (NiLuJe) [reply plugins/opds.koplugin/opdsbrowser.lua]: So, basically, what @Frenzie said, it makes sense in like 0.5% of the callers of this dialog ;p.

Comment (hius07) [reply plugins/opds.koplugin/opdsbrowser.lua]: Thanks, have reverted already.

File: plugins/calibre.koplugin/main.lua (hunk 5 above)
Comment (Frenzie) [reply plugins/calibre.koplugin/main.lua:112]: I slightly prefer calibre metadata search in the sense that it's more succinct, but no real preference

File: frontend/ui/widget/dictquicklookup.lua [hunk 12]
```diff
@@ -1199,7 +1199,11 @@ function DictQuickLookup:lookupInputWord(hint)
                     end,
                 },
                 {
-                    text = _("Lookup"),
+                    if self.is_wiki then
+                        text = _("Search Wikipedia")
+                    else
+                        text = _("Search dictionary")
+                    end,
```
Comment (NiLuJe) [inline frontend/ui/widget/dictquicklookup.lua:1206]: Can't do that, you're in a table list initializer ;).

```lua
text = self.is_wiki and _("Search Wikipedia") or _("Search dictionary"),
```

might work, though ;).

Comment (hius07) [reply frontend/ui/widget/dictquicklookup.lua:1206]: Thank you

File: plugins/calibre.koplugin/search.lua [hunk 13]
```diff
@@ -180,7 +180,7 @@ local CalibreSearch = InputContainer:new{
 
 function CalibreSearch:ShowSearch()
     self.search_dialog = InputDialog:new{
-        title = _("Search books"),
+        title = _("Enter string to search for"),
```
Comment (Frenzie) [inline plugins/calibre.koplugin/search.lua:183]: I prefer to keep programlish like `string` to the backend.

Comment (Frenzie) [reply plugins/calibre.koplugin/search.lua:183]: Since you like consistency ;-)
```suggestion
        title = _("Enter a word or phrase to look up"),
```

Comment (hius07) [reply plugins/calibre.koplugin/search.lua:183]: Seems 'look up' is not much applicable for calibre.
Maybe, return metadata?
`Enter text to search in metadata`
`Enter words to search in metadata`
Or get rid of "Enter...", just
`Calibre metadata search`

Comment (Frenzie) [reply plugins/calibre.koplugin/search.lua:183]: > Calibre metadata search

Sounds good to me, but pinging @NiLuJe since I don't use calibre.

Comment (NiLuJe) [reply plugins/calibre.koplugin/search.lua:183]: Yep, sounds good!

Comment (Frenzie) [reply plugins/calibre.koplugin/search.lua:183]: (for easy commit)
```suggestion
        title = _("Calibre metadata search"),
```

File: plugins/calibre.koplugin/main.lua (hunk 5 above)
Comment (Frenzie) [reply plugins/calibre.koplugin/main.lua:112]:
```suggestion
            text = _("Calibre metadata search"),
```

File: plugins/calibre.koplugin/main.lua [hunk 14]
```diff
@@ -59,7 +59,7 @@ function Calibre:closeWirelessConnection()
 end
 
 function Calibre:onDispatcherRegisterActions()
-    Dispatcher:registerAction("calibre_search", { category="none", event="CalibreSearch", title=_("Search in calibre metadata"), device=true,})
+    Dispatcher:registerAction("calibre_search", { category="none", event="CalibreSearch", title=_("Calibre search"), device=true,})
```
Comment (Frenzie) [inline plugins/calibre.koplugin/main.lua:62]:
```suggestion
    Dispatcher:registerAction("calibre_search", { category="none", event="CalibreSearch", title=_("Calibre metadata search"), device=true,})
```

**Comment by hius07** [comment]
Great, thanks!
